### PR TITLE
Hide referral in nightly

### LINKED ISSUE #11247: Remove `Recover` check for referral
# Problem
Currently users that recover an account are not eligible for a Starter Pack, even though they might be first time Status users. We cannot distinguish between new Status users and returning Status users based on use of recovering an account upon onboarding. 

## Implementation
Should be updated here: https://github.com/status-im/status-react/blob/develop/src/status_im/multiaccounts/login/core.cljs#L254-L256

## Acceptance Criteria
User receives prompt to Accept attribution (and Starter Pack depending on referral url type) when:
- Using a referral url 
- `Access existing key` > `Enter seed phrase` 
- This is the only multiaccount
- Is account creation flow

## Notes
Abuse by uninstalling and recovering the same account already used in Status is managed by IP check

### PR DESCRIPTION
**Pull request by Ferossgp**
Fixes #11247

### CODE REVIEW

**Review by rasom — APPROVED** [review]

File: src/status_im/multiaccounts/login/core.cljs [hunk 1]
```diff
@@ -250,9 +250,7 @@
   [{:keys [db] :as cofx}]
   (let [{:keys [multiaccount multiaccounts :multiaccount/accounts]} db
         {:keys [creating?]}                                         (:multiaccounts/login db)
-        recovering?                                                 (get-in db [:intro-wizard :recovering?])
```
Comment (flexsurfer) [inline src/status_im/multiaccounts/login/core.cljs:253]: is this related to this PR ?

Comment (Ferossgp) [reply src/status_im/multiaccounts/login/core.cljs:253]: Not related to nightly but this is also needed for the referrals program https://github.com/status-im/status-react/issues/11247

**Review by jakubgs — APPROVED** [review]

**Review by hesterbruikman — APPROVED** [review]
Thank you @Ferossgp 🙌 Can confirm that 1:1 invite (chat registration) and advertiser attribution request appears after recover multiaccount
